docs/templates/docs/search_results.html

```diff
@@ -20,17 +20,29 @@
     </search>
     <h2>
       {% if release.is_dev %}
-        {% blocktranslate count num_results=paginator.count trimmed %}
-          Only 1 result for <em>{{ query }}</em> in the development version
-        {% plural %}
-          {{ num_results }} results for <em>{{ query }}</em> in the development version
-        {% endblocktranslate %}
+        {% if paginator.count == 0 %}
+          {% blocktranslate trimmed %}
+            No results for <em>{{ query }}</em> in the development version
+          {% endblocktranslate %}
+        {% else %}
+          {% blocktranslate count num_results=paginator.count trimmed %}
+            Only 1 result for <em>{{ query }}</em> in the development version
```

Comment:
You should still be using `{{ num_result }}` in the singular form, in case we add support for a language where the singular form is used outside of 0 and 1 (you might also want to consider dropping the "only" for that same reason, but I'm not sure if that's as much of an issue):
```suggestion
            Only {{ num_results }} result for <em>{{ query }}</em> in the development version
```